repos/fdbt-site/src/pages/api/capValidity.ts

```diff
@@ -0,0 +1,90 @@
+import { NextApiResponse } from 'next';
+import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
+import {
+    MATCHING_JSON_ATTRIBUTE,
+    MATCHING_JSON_META_DATA_ATTRIBUTE,
+    CAP_EXPIRY_ATTRIBUTE,
+} from '../../constants/attributes';
+import { redirectToError, redirectTo, getAndValidateNoc } from '../../utils/apiUtils';
+import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
+import { getFareDayEnd } from '../../data/auroradb';
+import { CapExpiry, Ticket, WithIds } from '../../interfaces/matchingJsonTypes';
+import { putUserDataInProductsBucketWithFilePath } from '../../utils/apiUtils/userData';
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        const errors: ErrorInfo[] = [];
+        if (req.body.capValid) {
+            const { capValid } = req.body;
+
+            let productEndTime = '';
+            const endOfFareDay = await getFareDayEnd(getAndValidateNoc(req, res));
+
+            const ticket = getSessionAttribute(req, MATCHING_JSON_ATTRIBUTE);
+            const matchingJsonMetaData = getSessionAttribute(req, MATCHING_JSON_META_DATA_ATTRIBUTE);
+
+            if (capValid === 'fareDayEnd') {
+                if (!endOfFareDay) {
+                    errors.push({
+                        id: 'product-end-time',
+                        errorMessage: 'No fare day end defined',
+                    });
+
+                    updateSessionAttribute(req, CAP_EXPIRY_ATTRIBUTE, errors);
+                    redirectTo(res, '/selectCapValidity');
+
+                    return;
+                } else {
+                    productEndTime = endOfFareDay;
+                }
+            }
+
+            // redirected from the product details page
+            if (ticket && matchingJsonMetaData) {
```

Comment:
We dont want any of the edit code here, as we don't know if capped products will be able to be edited, or what shape they will be in.